### PR TITLE
Rework EmbeddingMaps to be compile time

### PR DESCRIPTION
**Pull request by nilsdeppe**
Embedding maps are now changed to "CoordinateMaps" so that when time-dependent maps are introduced they can be handle in effectively the same way. All embedding maps work in an "unknown" intermediate from, with only the wrapping `CoordinateMap` class actually being aware of the frames.

Note: I can move the maps from `Domain/EmbeddingMaps` to `Domain/CoordinateMaps` once we are happy with the code. I worry that renaming a directory would cause pain when squashing fixup commits.

### CODE REVIEW

**Comment by nilsdeppe** [comment]
Enable PUP test of Block

**Comment by nilsdeppe** [comment]
Remove commented out 3D identity test in Block

**Comment by nilsdeppe** [comment]
Block tests

File: src/DataStructures/Tensor/IndexType.hpp [hunk 1]
```diff
@@ -59,6 +62,10 @@ inline std::ostream& operator<<(std::ostream& os,
                                 const Frame::Distorted& /*meta*/) {
   return os << "Distorted";
 }
+inline std::ostream& operator<<(std::ostream& os,
```
Comment (nilsdeppe) [inline src/DataStructures/Tensor/IndexType.hpp:65]: test

File: src/Domain/EmbeddingMaps/CoordinateMap.hpp [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    for (size_t j = 0; j < Dim; ++j) {
+      out_jac.get(i, j) = std::move(jac.get(i, j));
+    }
+  }
+  return out_jac;
+}
+}  // namespace coordinate_map_detail
+
+/// \ingroup ComputationalDomain
+/// Base class representing a coordinate mapping from a
```
Comment (nilsdeppe) [inline src/Domain/EmbeddingMaps/CoordinateMap.hpp]: Update dox

File: src/Domain/EmbeddingMaps/CoordinateMap.hpp [hunk 3]
```diff
@@ -0,0 +1,262 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the abstract base class EmbeddingMap.
```
Comment (nilsdeppe) [inline src/Domain/EmbeddingMaps/CoordinateMap.hpp]: CoordinateMap

File: src/Domain/EmbeddingMaps/CoordinateMap.hpp [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                  SourceFrame>
+CoordinateMap<SourceFrame, TargetFrame, Maps...>::inverse(
+    const tnsr::I<T, dim, TargetFrame>& target_point) const {
+  std::array<T, dim> mapped_point = make_array<T, dim>(target_point);
+  tuple_fold<true>(maps_,
+                   [](const auto& map, std::array<T, dim>& point) {
+                     point = map.inverse(point);
+                   },
+                   mapped_point);
+  return Point<dim, SourceFrame>(std::move(mapped_point));
```
Comment (nilsdeppe) [inline src/Domain/EmbeddingMaps/CoordinateMap.hpp]: `tnsr::I`

File: src/Domain/EmbeddingMaps/CoordinateMap.hpp [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+    T, tmpl::integral_list<std::int32_t, 2, 1>,
+    index_list<
+        SpatialIndex<CoordinateMap<SourceFrame, TargetFrame, Maps...>::dim,
+                     UpLo::Up, SourceFrame>,
+        SpatialIndex<CoordinateMap<SourceFrame, TargetFrame, Maps...>::dim,
+                     UpLo::Lo, TargetFrame>>>
+CoordinateMap<SourceFrame, TargetFrame, Maps...>::inv_jacobian(
+    const tnsr::I<T, dim, SourceFrame>& source_point) const {
+  std::array<T, dim> mapped_point = make_array<T, dim>(source_point);
+  if (1 == sizeof...(Maps)) {
```
Comment (nilsdeppe) [inline src/Domain/EmbeddingMaps/CoordinateMap.hpp]: remove this block and handle it correctly below by moving `point = map(point)` into the `if (count != 0)`

File: src/Domain/EmbeddingMaps/Identity.hpp [hunk 6]
```diff
@@ -6,50 +6,64 @@
...
+class Identity {
  public:
+  static constexpr size_t dim = Dim;
+
   Identity() = default;
-  ~Identity() override = default;
-  Identity(const Identity&) = delete;
+  ~Identity() = default;
+  Identity(const Identity&) = default;
   Identity(Identity&&) noexcept = default;  // NOLINT
```
Comment (nilsdeppe) [inline src/Domain/EmbeddingMaps/Identity.hpp]: remove NOLINT

**Review by kidder — APPROVED** [review]

**Pull request merged**